src/parser/query_result.c

```diff
@@ -659,9 +660,11 @@ pt_get_titles (PARSER_CONTEXT * parser, PT_NODE * query)
 	}
       else
 	{
+	  save_custom = parser->custom_print;
 	  parser->custom_print |= PT_SUPPRESS_CHARSET_PRINT;
+	  parser->custom_print |= PT_PRINT_NO_CURRENT_USER_NAME;
 	  t = pt_get_node_title (parser, s, f);
-	  parser->custom_print &= ~PT_SUPPRESS_CHARSET_PRINT;
+	  parser->custom_print = save_custom;
```

Comment:
The semantics are slightly different from the existing code.
If the purpose of the existing code is preserved, it should be modified as follows.

```
	  parser->custom_print |= (PT_SUPPRESS_CHARSET_PRINT | PT_PRINT_NO_CURRENT_USER_NAME);
	  t = pt_get_node_title (parser, s, f);
	  parser->custom_print &= ~(PT_SUPPRESS_CHARSET_PRINT | PT_PRINT_NO_CURRENT_USER_NAME);
```

Reply:
The code to remove PT_SUPPRESS_CHARSET_PRINT may have been intentional, but I don't think it matters.
The previous code is misleading, so I'll keep the changes.